database/migrations/create_chat_tables.php

```diff
@@ -3,97 +3,89 @@
 use Illuminate\Database\Migrations\Migration;
 use Illuminate\Database\Schema\Blueprint;
 use Illuminate\Support\Facades\Schema;
+use Musonza\Chat\ConfigurationManager;
 
 class CreateChatTables extends Migration
 {
-    protected $userModelPrimaryKey = 'id';
-    protected $userModelTable = 'users';
-
-    protected $useBigIncrements;
-
-    public function __construct()
-    {
-        $config = config('musonza_chat');
-        $userModel = app($config['user_model']);
-
-        $this->userModelPrimaryKey = $userModel->getKeyName();
-        $this->userModelTable = $userModel->getTable();
-    }
-
     /**
      * Run the migrations.
      *
      * @return void
      */
     public function up()
     {
-        Schema::create('mc_conversations', function (Blueprint $table) {
+        Schema::create(ConfigurationManager::CONVERSATIONS_TABLE, function (Blueprint $table) {
             $table->bigIncrements('id');
             $table->boolean('private')->default(true);
+            $table->boolean('direct_message')->default(false);
             $table->text('data')->nullable();
             $table->timestamps();
         });
 
-        Schema::create('mc_messages', function (Blueprint $table) {
+        Schema::create(ConfigurationManager::MESSAGES_TABLE, function (Blueprint $table) {
             $table->bigIncrements('id');
             $table->text('body');
             $table->bigInteger('conversation_id')->unsigned();
-            $table->bigInteger('user_id')->unsigned();
+            $table->bigInteger('participation_id')->unsigned()->nullable();
             $table->string('type')->default('text');
             $table->timestamps();
 
-            $table->foreign('user_id')
-                    ->references($this->userModelPrimaryKey)
-                    ->on($this->userModelTable)
-                    ->onDelete('cascade');
+            $table->foreign('participation_id')
+                ->references('id')
+                ->on(ConfigurationManager::PARTICIPATION_TABLE)
+                ->onDelete('set null');
```

Comment:
I got this error trying to install on a mysql db ` General error: 1215 Cannot add foreign key constraint (SQL: alter table `chat_messages` add constraint `chat_messages_participation_id_foreign` foreign key (`participation_id`) references `chat_participation` (`id`) on delete set null)`

Reply:
I removed that part and it continue to migrate db but when i try to send a GET request to my endpoint with `$conversations = Chat::conversations()->setParticipant(auth()->user())->get();` it fails with 

```sql
General error: 3065 Expression #1 of ORDER BY clause is not in SELECT list, references column
```


But with `$conversations = Chat::conversations()->conversation->all();` it works, but `data is empty
<img width="711" alt="Skærmbillede 2019-11-02 kl  22 18 36" src="https://user-images.githubusercontent.com/183790/68077129-c4dde200-fdbe-11e9-94df-001981a0755a.png">